### DISCUSSION TITLE
DRILL TO DETAIL & DRILL BY permissions

### DISCUSSION THREAD

**Discussion by tsator** [apache/superset#28620] [Q&A / Help]
What permissions do i have to grant for the drill option to be displayed?  (Public & Gamma roles)

- Permission **can_explore_on_Superset** must be disabled to prevent them from edititing the graph  

- **Currently these flags are enabled:** 
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; - "DRILL_TO_DETAIL": True,
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; - "DRILL_BY": True,

- **Some of the permissions granted:**  
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can_samples_on_Datasource
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can_write_ExploreFormDataRestAPI
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can_read_on_Explore
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can_read_on_Dashboard
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can_read_on_Chart
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can read on DashboardFilterStateRestApi
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can write on DashboardFilterStateRestApi
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can fetch datasource metadata on Superset
&emsp; &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;  - can read on Dataset

I tried to do what is explained [here](https://github.com/apache/superset/pull/27029) but it didnt work

**Comment by emwaka14** [comment]
I am facing the same issue and wondering if there is a fix for it. 

Actually, I intend for the dashboard to be published and accessible by the public (anyone with the permalink) and they must be able to drill to detail. 

But if it is not possible, then I could settle with a user group that has no edit access but can drill to detail--I could then find my own way for authentication and access. I hope we finally get a solution to this.
